### DISCUSSION TITLE
Re-sizing a JPG to a target number of Pixels.

### DISCUSSION THREAD

**Discussion by AlanGitMan** [ImageMagick/ImageMagick#2638] [Help] [answered]
Hi everyone,

I have a digital picture frame. It has a limitation that pictures must not exceed a set pixel count (about 910K 24-bit pixels). 

What I would like is to have a simple way to run a convert command over a number of images, scaling them so that they don't exceed this pixel limit - while still maintaining their aspect ratios.  However, the images concerned come from all over the place and widly vary in size  -though they ARE all JPG. 

I have looked though the convert command's options but the scaling options all seem to be about scaling to an x or a y target number - not a total pixel count.   I could write a Bash script to repeatedly use convert to labriously extract the x and y size, then again to figure out which is the largest dimension and then do the sums to scale it close to my target limit.  

However, since the photo frame concept was quite popular for a while, I am hoping I am not the first person to have this need.  Therefore, before I start scripting, I wanted to make sure that there is nothing in the IM toolkit that I missed which can already do this for me? 

Thanks for any help or suggestions

Alan T

**Comment by snibgo** [comment, marked as answer]
See https://imagemagick.org/script/command-line-processing.php#geometry , search for "area@". Give your desired upper limit in pixels, suffixed with "@". For example:
```
magick toes.png -size 1000@ x.png
```
The result is 33x29 pixels, which is 957 pixels.

**Reply by AlanGitMan** [reply]
Exactly what I need - 100% solve my problem - many thanks for your help. 

One small thing, I am using an older version of IM on my Ubuntu system and so used the "convert" command with -resize option to get this done - like...

   convert incoming.jpg -resize 918000@ outfile.jpg

and to use "convert" to a resize a batch of jpegs in "sourcefolder" to output files in destination subfolder,  I used the command sequence:

   cd /home/sourcefolder
   mkdir destfolder
   for x in `ls -1 *.jpg` ; do convert "$x" -resize 918000@ "./destfolder/$x" ; echo "$x resized" ; done ; echo "COMPLETED"

Thanks again for the help on this - IM never ceases to impress me!

**Comment by fmw42** [comment]
See -define jpeg:extent={size} at https://imagemagick.org/Usage/formats/#jpg_write to compress to a specific file size in kb
